src/importlinter/application/contract_utils.py

```diff
@@ -46,6 +46,13 @@ def remove_ignored_imports(
     return warnings
 
 
+def resolve_module_expressions(
+    graph: ImportGraph, expressions: Iterable[ModuleExpression]
+) -> Iterable[Module]:
```

Comment:
I wonder if it would be better to return a `set[Module]` here?

Otherwise we might want to document the expected behaviour with respect to (i) ordering and (ii) duplicates.

Reply:
Good idea. The set would make much more sense semantically. The performance might be a little worse, but I doubt that it would be noticeable.